### PR TITLE
refactor(elixir): session handshake and worker option names

### PR DESCRIPTION
**Pull request by davidag**
## Current Behavior

NA

## Proposed Changes

Apply a variable name refactor as suggested in https://github.com/build-trust/ockam/issues/3653

I'm not familiar with the code, but I'd like to make sure that you want to rename `worker_mod` to `data_mod`. I was wondering if `data_worker_mod` could be more meaningful. What do you think @hairyhum ?

## Checks

- [x] All commits in this Pull Request are [signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) and Verified by Github.
- [x] All commits in this Pull Request follow the Ockam [commit message convention](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages).
- [x] I accept the Ockam Community [Code of Conduct](https://github.com/build-trust/.github/blob/main/CODE_OF_CONDUCT.md).
- [x] I have accepted the Ockam [Contributor License Agreement](https://github.com/build-trust/ockam-contributors/blob/main/CLA.md) by adding my Git/Github details in a row at the end of the [CONTRIBUTORS.csv](https://github.com/build-trust/ockam-contributors/blob/main/CONTRIBUTORS.csv) file in a separate pull request to the [build-trust/ockam-contributors](https://github.com/build-trust/ockam-contributors) repository.

### CODE REVIEW

**Comment by hairyhum** [comment]
@davidag Yes, I think `data_worker_mod` makes more sense.

**Comment by davidag** [comment]
@hairyhum done, let me know if I'm missing something. thanks! 👌🏼

File: implementations/elixir/ockam/ockam/lib/ockam/session/spawner.ex [hunk 1]
```diff
@@ -13,14 +13,14 @@ defmodule Ockam.Session.Spawner do
     @spec spawner_setup(options :: Keyword.t(), state :: map) :: {:ok, options :: Keyword.t(), state :: map()} | {:error, reason :: any()}
     defaults to `&default_spawner_setup`
 
-  Upon receiving a message, `worker_mod` worker will be started
+  Upon receiving a message, `data_worker_mod` worker will be started
```
Comment (hairyhum) [inline implementations/elixir/ockam/ockam/lib/ockam/session/spawner.ex]: Spawner does not have a handshake and data stages, so this should be just `worker_mod`

**Review by hairyhum — CHANGES_REQUESTED** [review]

File: implementations/elixir/ockam/ockam/lib/ockam/session/spawner.ex (hunk 1 above)
Comment (davidag) [reply implementations/elixir/ockam/ockam/lib/ockam/session/spawner.ex]: I think it's fixed now 🤔

**Comment by hairyhum** [comment]
@davidag Looks good. Can you please address comments in the CLA PR https://github.com/build-trust/ockam-contributors/pull/151 before we can merge the PRs?

**Comment by davidag** [comment]
> @davidag Looks good. Can you please address comments in the CLA PR [build-trust/ockam-contributors#151](https://github.com/build-trust/ockam-contributors/pull/151) before we can merge the PRs?

@hairyhum done, also fixed the formatting error and rebased.

**Comment by hairyhum** [comment]
Thanks.
@davidag can you also squash your commits, make sure the resulting commit follows the [format](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages) and make sure that it's signed by a signature verified by github?

**Comment by davidag** [comment]
> Thanks. @davidag can you also squash your commits, make sure the resulting commit follows the [format](https://github.com/build-trust/.github/blob/main/CONTRIBUTING.md#commit-messages) and make sure that it's signed by a signature verified by github?

@hairyhum Done! 👌🏼

**Comment by davidag** [comment]
@hairyhum fixed conflicts with `develop`.

**Review by hairyhum — DISMISSED** [review]

**Review by hairyhum — APPROVED** [review]

**Pull request merged**